pkg/apis/servicecatalog/v1alpha1/defaults_test.go

```diff
@@ -68,6 +70,64 @@ func roundTrip(t *testing.T, obj runtime.Object) runtime.Object {
 	return obj3
 }
 
+func TestSetDefaultServiceBroker(t *testing.T) {
+	cases := []struct {
+		name     string
+		broker   *versioned.ServiceBroker
+		behavior versioned.ServiceBrokerRelistBehavior
+		duration *metav1.Duration
+	}{
+		{
+			name:     "neither duration or behavior set",
+			broker:   &versioned.ServiceBroker{},
+			behavior: versioned.ServiceBrokerRelistBehaviorDuration,
+			duration: &metav1.Duration{15 * time.Minute},
+		},
+		{
+			name: "behavior set to manual",
+			broker: func() *versioned.ServiceBroker {
+				b := &versioned.ServiceBroker{}
+				b.Spec.RelistBehavior = versioned.ServiceBrokerRelistBehaviorManual
+				return b
+			}(),
+			behavior: versioned.ServiceBrokerRelistBehaviorManual,
+			duration: nil,
+		},
+		{
+			name: "behavior set to duration but no duration provided",
+			broker: func() *versioned.ServiceBroker {
+				b := &versioned.ServiceBroker{}
+				b.Spec.RelistBehavior = versioned.ServiceBrokerRelistBehaviorDuration
+				return b
+			}(),
+			behavior: versioned.ServiceBrokerRelistBehaviorDuration,
+			duration: &metav1.Duration{15 * time.Minute},
+		},
+	}
+
+	for _, tc := range cases {
+		o := roundTrip(t, runtime.Object(tc.broker))
+		ab := o.(*versioned.ServiceBroker)
+		actualSpec := ab.Spec
+
+		if tc.behavior != actualSpec.RelistBehavior {
+			t.Errorf(
+				"%v: unexpected default RelistBehavior: expected %v, got %v",
+				tc.name, tc.behavior, actualSpec.RelistBehavior,
+			)
+		}
+
+		if tc.duration == nil && actualSpec.RelistDuration == nil {
```

Comment:
This check does not verify that the actual RelistDuration was nil for the "behavior set to manual" test.

Reply:
There is an open question as to whether or not RelistDuration MUST BE nil given a RelistBehavior of Manual. Is this something that needs to be asserted? CC @pmorie

Reply:
@eriknelson yes, it should be enforced to nil (and checked at the validation stage). similar to union types, where only one branch can be initialized at a time.